### PR TITLE
fix: add `PreferredDisplayName` to user_model

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

- excludes logic of resolving preferred display name to shared `contacts_utils`
- expose `PreferredDisplayName` role from `user_model`
- improves `PreferredDisplayName` handling in `member_model`
- improves granularity of data change reporting in `user_model`

Now the logic is kept in dedicated method (so far located in data method of `member_model`). It allows to reuse them in various context - in `data` methods, for updates and also other models like `user_model`.

Closes: #16429

### Affected areas

`member_model`, `user_model`

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Architecture guidelines](../architecture-guide.md)

### Screenshot of functionality (including design for comparison)

image

image

### CODE REVIEW

**Review by osmaczko — APPROVED** [review]
Good to see you in NIM layer :wave:

**Comment by anastasiyaig** [comment]
That works now https://ci.status.im/job/status-desktop/job/e2e/job/manual/2527/allure/

**Review by anastasiyaig — APPROVED** [review]

**Comment by anastasiyaig** [comment]
I've added a change to existing test of 1x1 chat to accept contact requests through settings now so this issue is now covered on PRs level
I also enabled group chat test in PRs, which adds contacts in activity center

Screenshot 2024-10-01 at 16 23 23

File: src/app/modules/shared_models/user_model.nim [hunk 1]
```diff
@@ -300,25 +316,46 @@ QtObject:
...
+    if self.items[ind].icon != icon:
+      self.items[ind].icon = icon
+      roles.add(ModelRole.Icon.int)
+
+    if self.items[ind].isUntrustworthy != isUntrustworthy:
+      self.items[ind].isUntrustworthy = isUntrustworthy
+      roles.add(ModelRole.IsUntrustworthy.int)
+
+    if preferredDisplayNameChanged:
+      roles.add(ModelRole.PreferredDisplayName.int)
```
Comment (jrainville) [inline src/app/modules/shared_models/user_model.nim:355]: Missing the return if no roles changed

**Review by jrainville — COMMENTED** [review]
Looks good, just two little missing stuff

**Review by jrainville — APPROVED** [review]
Nice job. Sorry about the regression

**Review by caybro — APPROVED** [review]
LGTM!

**Pull request merged**